src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -26,6 +27,28 @@ public sealed partial class NpgsqlReadBuffer
 
         internal Stream Underlying { private get; set; }
 
+        CancellationTokenSource _timeoutCts = new CancellationTokenSource();
+
+        internal int ReadTimeout
```

Comment:
nit: rename to Timeout (since we're already on NpgsqlReadBuffer), same with _readTimeout.